servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherAsBlockingIterable.java

```diff
@@ -103,8 +104,10 @@
         private boolean terminated;
 
         SubscriberAndIterator(int queueCapacity) {
-            maxBufferedItems = queueCapacity;
-            itemsToNextRequest = max(1, maxBufferedItems / 2);
+            itemsToNextRequest = max(1, queueCapacity / 2);
+            // We set this value ahead of time to avoid modifications from different threads. This will be true as
+            // soon as onSubscribe is called.
+            itemsRequested = itemsToNextRequest;
             this.queueCapacity = queueCapacity;
             data = new LinkedBlockingQueue<>();
```

Comment:
@Scottmitch while I'm here, does it make sense to switch to `ArrayBlockingQueue` (because the queue capacity is known in advance and is has an upper bound of 128), account for one more terminal event, and adjust `itemsToNextRequest` to max(1, queueCapacity - 1)`?